File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.py
class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        ans = ['a' for _ in range(n)]
        k -= n
        j = n-1
        while k > 0:
            ans[j] = chr(min(25, k) + ord('a'))
            k -= min(25, k)
            j -= 1
        return ''.join(ans)

File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.java
class Solution {
    public String getSmallestString(int n, int k) {
        char[] result = new char[n];
        for(int i=0; i<n; i++)
            result[i] = 'a';
        k -= n;
        
        while(k>0) {
            result[--n] = (char)(Math.min(25, k) + 'a');
            k -= Math.min(25, k);
        }
        return new String(result);
    }
}